add script to generate v3 ea

karen-stepanyan opened the pull request:
## Description

EA framework now provides interactive tool to generate new v3 adapters. This PR adds  a functionality to use that tool in monorepo. 

## Changes

- two new devDeps in main package.json, `@chainlink/external-adapter-framework` and `yo`
- `new` script is changed to use ea-framework for adapter generation
- packages/examples is removed
- Docs are modified 


## Steps to Test
To create test adapter, run 

`yarn new` 

optionally you can specify also the type (source is the default)

`yarn new composite`

alejoberardino reviewed (changes requested):
Only change would be to directly remove the previous generation script and keep this as the only one. If preferred, this PR could just change the reference and we could open a separate PR to remove the old v2 generation

karen-stepanyan commented:
hey @alejoberardino . Yeah, i think it  makes sense to replace it here so that we don't have a new script. 
I also deleted examples folder and updated the contributing file

alejoberardino reviewed (approved):
Approved, left one comment that can be ignored if it's not relevant

jacksnipe24 commented on CONTRIBUTING.md:
````diff
@@ -17,37 +17,28 @@ Thank you for your interest in improving the Chainlink External Adapter codebase
+$ yarn new source
 ```
 
 _If on a Mac, this requires `gnu-sed` to be installed and set as the default for the command `sed`._
 
 You can open a PR with the [New EA PR Template](./.github/PULL_REQUEST_TEMPLATE/new_ea_pr_template.md) by replacing `<branch>` in this URL: [https://github.com/smartcontractkit/external-adapters-js/compare/develop...<branch>?quick_pull=1&template=new_ea_pr_template.md](https://github.com/smartcontractkit/external-adapters-js/compare/develop...<branch>?quick_pull=1&template=new_ea_pr_template.md)
````
I think not only CONTRIBUTING.md but also other doc files need to be reviewed and updated. At least the PR opening checklist is not up to date at this point and also `gnu-sed` seems is not installed on my machine and it's not a problem at all.  Not sure if this is a right PR for them tho.

karen-stepanyan replied:
I think not only CONTRIBUTING.md but also other doc files need to be reviewed and updated. At least the PR opening checklist is not up to date at this point and also gnu-sed seems is not installed on my machine and it's not a problem at all. Not sure if this is a right PR for them tho.
